### ISSUE TITLE
Crash ``Parsing Python code failed: unexpected indent (<unknown>, line 2)`` using  disable-next=duplicate-code

### ISSUE THREAD

**Issue by parched** [PyCQA/pylint#6496]
### Bug description

When parsing the following file:
```python
# pylint: disable-next=duplicate-code
a_list = [
    "a",
]

```

pylint crashed with a ``AstroidSyntaxError`` and with the following stacktrace:
```
Traceback (most recent call last):
  File "C:\Users\me\AppData\Local\pypoetry\Cache\virtualenvs\my-project-UjAHidfa-py3.10\lib\site-packages\astroid\builder.py", line 168, in _data_build
    node, parser_module = _parse_string(data, type_comments=True)
  File "C:\Users\me\AppData\Local\pypoetry\Cache\virtualenvs\my-project-UjAHidfa-py3.10\lib\site-packages\astroid\builder.py", line 454, in _parse_string
    parsed = parser_module.parse(data + "\n", type_comments=type_comments)
  File "C:\Users\me\AppData\Local\pypoetry\Cache\virtualenvs\my-project-UjAHidfa-py3.10\lib\site-packages\astroid\_ast.py", line 49, in parse
    return parse_func(string)
  File "C:\Users\me\AppData\Local\Programs\Python\Python310\lib\ast.py", line 50, in parse
    return compile(source, filename, mode, flags,
  File "<unknown>", line 2
    "a",
IndentationError: unexpected indent

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\me\AppData\Local\pypoetry\Cache\virtualenvs\my-project-UjAHidfa-py3.10\lib\site-packages\pylint\lint\pylinter.py", line 1111, in _check_files
    self._check_file(get_ast, check_astroid_module, file)
  File "C:\Users\me\AppData\Local\pypoetry\Cache\virtualenvs\my-project-UjAHidfa-py3.10\lib\site-packages\pylint\lint\pylinter.py", line 1146, in _check_file
    check_astroid_module(ast_node)
  File "C:\Users\me\AppData\Local\pypoetry\Cache\virtualenvs\my-project-UjAHidfa-py3.10\lib\site-packages\pylint\lint\pylinter.py", line 1298, in check_astroid_module
    retval = self._check_astroid_module(
  File "C:\Users\me\AppData\Local\pypoetry\Cache\virtualenvs\my-project-UjAHidfa-py3.10\lib\site-packages\pylint\lint\pylinter.py", line 1341, in _check_astroid_module
    checker.process_module(node)
  File "C:\Users\me\AppData\Local\pypoetry\Cache\virtualenvs\my-project-UjAHidfa-py3.10\lib\site-packages\pylint\checkers\similar.py", line 832, in process_module
    self.append_stream(self.linter.current_name, stream, node.file_encoding)  # type: ignore[arg-type]
  File "C:\Users\me\AppData\Local\pypoetry\Cache\virtualenvs\my-project-UjAHidfa-py3.10\lib\site-packages\pylint\checkers\similar.py", line 373, in append_stream
    LineSet(
  File "C:\Users\me\AppData\Local\pypoetry\Cache\virtualenvs\my-project-UjAHidfa-py3.10\lib\site-packages\pylint\checkers\similar.py", line 663, in __init__
    self._stripped_lines = stripped_lines(
  File "C:\Users\me\AppData\Local\pypoetry\Cache\virtualenvs\my-project-UjAHidfa-py3.10\lib\site-packages\pylint\checkers\similar.py", line 566, in stripped_lines
    tree = astroid.parse("".join(lines))
  File "C:\Users\me\AppData\Local\pypoetry\Cache\virtualenvs\my-project-UjAHidfa-py3.10\lib\site-packages\astroid\builder.py", line 281, in parse
    return builder.string_build(code, modname=module_name, path=path)
  File "C:\Users\me\AppData\Local\pypoetry\Cache\virtualenvs\my-project-UjAHidfa-py3.10\lib\site-packages\astroid\builder.py", line 138, in string_build
    module, builder = self._data_build(data, modname, path)
  File "C:\Users\me\AppData\Local\pypoetry\Cache\virtualenvs\my-project-UjAHidfa-py3.10\lib\site-packages\astroid\builder.py", line 170, in _data_build
    raise AstroidSyntaxError(
astroid.exceptions.AstroidSyntaxError: Parsing Python code failed:
unexpected indent (<unknown>, line 2)
```

I worked around thing by using `disable` instead of `disable-next` inside the list.

### Command used

```shell
poetry run pylint a.py
```

### Pylint output

```shell
************* Module a
a.py:1:0: F0002: a.py: Fatal error while checking 'a.py'. Please open an issue in our bug tracker so we address this. There is a pre-filled template that you can use in 'C:\Users\me\AppData\Local\pylint\pylint\Cache\pylint-crash-2022-05-04-14.txt'. (astroid-error)
```

### Expected behavior

No error

### Pylint version

```shell
pylint 2.13.4
astroid 2.11.2
Python 3.10.4 (tags/v3.10.4:9d38120, Mar 23 2022, 23:13:41) [MSC v.1929 64 bit (AMD64)]
```

### OS / Environment

Windows 10

**Comment by jacobtylerwalls** [comment]
Duplicate of #6301, fixed in 2.13.6 (let us know if upgrading does not resolve your issue!)

**Comment by parched** [comment]
Thanks, that indeed fixes it. I should have checked I was using the latest version.
